### PR TITLE
Apply various improvements to `@QuarkusIntegrationTest`

### PR DESCRIPTION
**Pull request by geoand**
* Add argLine to all IT launchers and clean up related legacy configuration
* Fix issue with wait time in @QuarkusIntegrationTest
* Introduce new SPI for custom startup handling in @QuarkusIntegrationTest

Relates to #18399 and #17560 but doesn't quite close them as I have a few more things in mind for a follow up, plus I need to add documentation about all these features

### CODE REVIEW

**Comment by geoand** [comment]
@stuartwdouglas wanna take a look at this as this could easily have conflicts soon :)

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 1]
```diff
@@ -322,13 +322,6 @@ Executing [/data/home/gsmet/git/quarkus-quickstarts/getting-started/target/getti
 ...
 ----
 
-[TIP]
-====
-By default, Quarkus waits for 60 seconds for the native image to start before automatically failing the native tests. This
```
Comment (stuartwdouglas) [inline docs/src/main/asciidoc/building-native-image.adoc:327]: Why did you remove this section? Shouldn't it have just been updated to the new name?

Comment (geoand) [reply docs/src/main/asciidoc/building-native-image.adoc:327]: You are right. I'll fix it

**Review by stuartwdouglas — APPROVED** [review]
